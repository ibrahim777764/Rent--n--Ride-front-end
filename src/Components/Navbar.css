:root {
  --main-color: #96be0f;
  --main-font-family: "Archivo Black", sans-serif;
}

* {
  color: black;
}

.active {
  background-color: var(--main-color);
  color: white;
}

.navbar {
  display: flex;
  max-width: 175px;
  flex-direction: column;
  justify-content: space-between;
  height: 97vh;
  text-align: center;
  background-color: white;
}

.menu-header {
  text-align: center;
  margin-top: 2rem;
  font-size: 1.5rem;
  font-family: "Cookie", cursive;
  text-decoration: underline;
  transform: rotateZ(-10deg);
}

nav {
  display: flex;
  justify-content: right;
  font-family: var(--main-font-family);
  font-size: 15px;
}

.nav-links {
  width: 85%;
}

.nav-links li {
  text-align: left;
}

.nav-links a {
  padding: 10px 15px;
  display: block;
}

.media-links {
  display: flex;
  justify-content: center;
}
.media-links li {
  padding: 2px;
}
