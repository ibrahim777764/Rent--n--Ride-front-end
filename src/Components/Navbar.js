import React from 'react'
import { Link, NavLink } from 'react-router-dom'
import * as FaIcons from "react-icons/fa";
// import './Navbar.css'
import '../Helpers/ComponentsHelper'

function Navbar() {
  return (
    <>
      <div className="navbar">
        <Link to="/" className="menu-header">
          RENT & RIDE
        </Link>
        <nav>
          <ul className='nav-links'>
            <li>
              <NavLink to="/models" >MODELS</NavLink>
            </li>
            <li>
              <NavLink to="/reserve">RESERVE</NavLink>
            </li>
            <li>
              <NavLink to="/my_reservations">MY RESERVATIONS</NavLink>
            </li>
            <li>
              <NavLink to="/add_car">ADD CAR</NavLink>
            </li>
            <li>
              <NavLink to="/delete_car">DELETE CAR</NavLink>
            </li>
          </ul>
        </nav>
        <div className="navbar-footer">
          <ul className='media-links'>
            <li>
              <Link to="/">
                <FaIcons.FaTwitter />
              </Link>
            </li>
            <li>
              <Link to="/">
                <FaIcons.FaFacebookF />
              </Link>
            </li>
            <li>
              <Link to="/">
                <FaIcons.FaGooglePlusG />
              </Link>
            </li>
            <li>
              <Link to="/">
                <FaIcons.FaVimeoV />
              </Link>
            </li>
            <li>
              <Link to="/">
                <FaIcons.FaPinterestP />
              </Link>
            </li>
          </ul>
          <p>Copyright 2022</p>
        </div>
      </div>
    </>
  )
}

export default Navbar