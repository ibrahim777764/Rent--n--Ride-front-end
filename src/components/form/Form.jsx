import PropTypes from 'prop-types';

const Form = (props) => {
  const { data } = props;

  console.log(data);

  return (
    <fragment>
      <div className="grid-display grid-simple">
        <div className="add-margin-below">
          <label htmlFor="brand">Brand</label>
          <input
            type="text"
            id="brand"
            name="brand"
            placeholder="Ferrari"
            className="form-field"
            required
          />
        </div>
      </div>

      <div className="grid-display grid-double">
        <div className="add-margin-below">
          <label htmlFor="brand">Model</label>
          <input
            type="text"
            id="model"
            name="model"
            placeholder="Testarossa"
            className="form-field"
            required
          />
        </div>

        <div className="add-margin-below grid-double">
          <label htmlFor="year">Year</label>
          <input
            type="text"
            id="year"
            name="year"
            placeholder="2018"
            className="form-field"
            required
          />
        </div>
      </div>

      <div className="grid-display grid-double">
        <div className="add-margin-below grid-double">
          <label htmlFor="country">Country</label>
          <input
            type="text"
            id="country"
            name="country"
            placeholder="Italia"
            className="form-field"
            required
          />
        </div>

        <div className="add-margin-below">
          <label htmlFor="vehicle">Color</label>
          <select
            name="vehicle"
            className="form-field"
            // defaultValue={vehicle || ''}
            required
          >
            <option value="">Select one</option>
            <option value="Black">Black</option>
            <option value="Blue">Blue</option>
            <option value="Brown">Brown</option>
            <option value="Carbon">Carbon</option>
            <option value="Green">Green</option>
            <option value="Orange">Orange</option>
            <option value="Red">Red</option>
            <option value="Silver">Silver</option>
            <option value="Yellow">Yellow</option>
            <option value="White">White</option>
          </select>
        </div>
      </div>

      <h3>Technical Specs</h3>

      <div className="grid-display grid-triple">
        <div className="add-margin-below">
          <label htmlFor="power">Power</label>
          <input
            type="text"
            id="power"
            name="power"
            placeholder="780 CV (574 kW)"
            className="form-field"
            required
          />
        </div>

        <div className="add-margin-below">
          <label htmlFor="max_speed">Max Speed</label>
          <input
            type="text"
            id="max_speed"
            name="max_speed"
            placeholder="355 km/h"
            className="form-field"
            required
          />
        </div>

        <div className="add-margin-below">
          <label htmlFor="acceleration">Acceleration</label>
          <input
            type="text"
            id="acceleration"
            name="acceleration"
            placeholder="9.8s"
            className="form-field"
            required
          />
        </div>
      </div>

      <h3>Description</h3>

      <div className="add-margin-below">
        <textarea
          name="description"
          className="form-field"
          required
        />
      </div>

    </fragment>
  );
};

Form.propTypes = {
  data: PropTypes.array,
};

Form.defaultProps = {
  data: null,
};

export default Form;
