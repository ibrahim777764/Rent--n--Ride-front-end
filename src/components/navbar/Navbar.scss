@import "../../config.scss";

.navbar {
  display: none;
}

// #######################################################################################
// ################################### DESKTOP VERSION ###################################
// #######################################################################################

@media (min-width: $breakpoint-tablet) {
  nav {
    width: 90%;
    font-weight: 700;
    font-size: 1.15rem;

    ul {
      list-style-type: none;
    }

    a {
      background-color: $color-white;
      border-right: 1px solid $color-silver;
      color: $color-dark;
      display: block;
      margin-right: -2px;
      padding: $padding * 1.25;
      padding-bottom: $padding * 0.9;
      text-decoration: none;
      transition: $transition;

      &:hover {
        background-color: $color-silver;
        border-right: 1px solid $color-silver;
        text-decoration: none;
        transition: $transition;
      }
    }

    .active,
    .active:hover {
      background-color: $color-primary;
      border-right: 1px solid $color-primary;
      color: $color-white;
    }
  }

  .img-container {
    background: url(../../assets/images/logo.png) no-repeat;
    background-position: center center;
    background-size: auto 35%;
    align-items: center;
    display: flex;
    height: $navbar-size;
    justify-content: center;
    margin-bottom: 5vh;
    transition: $transition;
    width: $navbar-size;

    &:hover {
      opacity: 0.5;
      transition: $transition;
    }
  }

  .navbar {
    background-color: $color-white;
    border-right: 1px solid $color-silver;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: space-between;
    left: 0;
    position: fixed;
    top: 0;
    width: $navbar-size;
    z-index: 1000;
  }

  .navTop {
    align-items: flex-end;
    display: flex;
    flex-direction: column;
  }

  .social-icons {
    padding: $padding;
    text-align: center;
    width: 100%;

    ul li {
      display: inline-block;
      margin: 0 $padding/2;
    }

    p {
      font-size: 0.75rem;
    }
  }
}
