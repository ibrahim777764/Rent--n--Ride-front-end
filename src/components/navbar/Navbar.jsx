/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable no-script-url */
/* eslint linebreak-style: ["error", "windows"] */
import React, { useState, useEffect } from 'react';
import { NavLink, Link } from 'react-router-dom';
import {
  FaBars,
  FaFacebookF,
  FaGooglePlusG,
  FaPinterestP,
  FaTimes,
  FaTwitter,
} from 'react-icons/fa';
import './Navbar.scss';
import Button from '../button/Button';

const Navbar = () => {
  const [navVisible, setNavVisible] = useState(true);
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isAdmin, setIsAdmin] = useState(false);

  useEffect(() => {
    if (window.innerWidth < 400) {
      setNavVisible(false);
    }

    if (JSON.parse(localStorage.getItem('current_user'))) {
      setIsLoggedIn(true);
      if (JSON.parse(localStorage.getItem('current_user')).role === 'admin') {
        setIsAdmin(true);
      } else {
        setIsAdmin(false);
      }
    } else {
      setIsLoggedIn(false);
    }
  }, []);

  const endSession = () => {
    window.localStorage.removeItem('current_user');
    window.localStorage.removeItem('token');
    window.location.href = '/login';
  };

  return (
    <>
      <div className="menuMobile" onClick={() => setNavVisible(!navVisible)}>
        <FaBars className="icon" />
      </div>
      <div className={navVisible ? 'navbar nav-show' : 'navbar nav-hide'}>
        <div
          className="closeBtn"
          onClick={() => setNavVisible(false)}
        >
          <FaTimes className="icon" />
        </div>

        <div className="navTop">
          <Link to="/" className="img-container" onClick={() => setNavVisible(!navVisible)} />

          <nav>
            <ul>
              <li>
                <NavLink to="/models" onClick={() => setNavVisible(false)}>
                  MODELS
                </NavLink>
              </li>
              <li>
                <NavLink to="/reservation" onClick={() => setNavVisible(false)}>
                  MY RESERVATIONS
                </NavLink>
              </li>
              <li>
                <NavLink to="/booking" onClick={() => setNavVisible(false)}>
                  RESERVATIONS
                </NavLink>
              </li>
              <li>
                <NavLink to="/about" onClick={() => setNavVisible(false)}>
                  ABOUT
                </NavLink>
              </li>
              {isAdmin && (
              <li>
                <NavLink to="/admin" onClick={() => setNavVisible(false)}>
                  ADMIN
                </NavLink>
              </li>
              )}
              {isLoggedIn && (
              <li className="logout-button">
                <Button
                  label="LOGOUT"
                  size="normal"
                  color="dark"
                  btnAxn={endSession}
                />
              </li>
              )}

            </ul>
          </nav>
        </div>

        <div className="social-icons">
          <ul>
            <li><FaTwitter className="icon" /></li>
            <li><FaFacebookF className="icon" /></li>
            <li><FaGooglePlusG className="icon" /></li>
            <li><FaPinterestP className="icon" /></li>
          </ul>

          <p>Microverse Copyright 2022</p>
        </div>
      </div>
    </>
  );
};

export default Navbar;
