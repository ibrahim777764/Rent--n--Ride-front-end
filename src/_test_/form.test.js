import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Form from '../components/form/Form';
import Navbar from '../components/navbar/Navbar';

it('Test Form renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Form />
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
