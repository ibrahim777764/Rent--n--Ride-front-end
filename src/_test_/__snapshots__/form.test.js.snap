// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Form renders correctly 1`] = `
Array [
  <fragment>
    <div
      className="grid-display grid-simple"
    >
      <div
        className="add-margin-below"
      >
        <label
          htmlFor="brand"
        >
          Brand
        </label>
        <input
          className="form-field"
          id="brand"
          name="brand"
          placeholder="Ferrari"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="grid-display grid-double"
    >
      <div
        className="add-margin-below"
      >
        <label
          htmlFor="brand"
        >
          Model
        </label>
        <input
          className="form-field"
          id="model"
          name="model"
          placeholder="Testarossa"
          required={true}
          type="text"
        />
      </div>
      <div
        className="add-margin-below grid-double"
      >
        <label
          htmlFor="year"
        >
          Year
        </label>
        <input
          className="form-field"
          id="year"
          name="year"
          placeholder="2018"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="grid-display grid-double"
    >
      <div
        className="add-margin-below grid-double"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <input
          className="form-field"
          id="country"
          name="country"
          placeholder="Italia"
          required={true}
          type="text"
        />
      </div>
      <div
        className="add-margin-below"
      >
        <label
          htmlFor="vehicle"
        >
          Color
        </label>
        <select
          className="form-field"
          name="vehicle"
          required={true}
        >
          <option
            value=""
          >
            Select one
          </option>
          <option
            value="Black"
          >
            Black
          </option>
          <option
            value="Blue"
          >
            Blue
          </option>
          <option
            value="Brown"
          >
            Brown
          </option>
          <option
            value="Carbon"
          >
            Carbon
          </option>
          <option
            value="Green"
          >
            Green
          </option>
          <option
            value="Orange"
          >
            Orange
          </option>
          <option
            value="Red"
          >
            Red
          </option>
          <option
            value="Silver"
          >
            Silver
          </option>
          <option
            value="Yellow"
          >
            Yellow
          </option>
          <option
            value="White"
          >
            White
          </option>
        </select>
      </div>
    </div>
    <h3>
      Technical Specs
    </h3>
    <div
      className="grid-display grid-triple"
    >
      <div
        className="add-margin-below"
      >
        <label
          htmlFor="power"
        >
          Power
        </label>
        <input
          className="form-field"
          id="power"
          name="power"
          placeholder="780 CV (574 kW)"
          required={true}
          type="text"
        />
      </div>
      <div
        className="add-margin-below"
      >
        <label
          htmlFor="max_speed"
        >
          Max Speed
        </label>
        <input
          className="form-field"
          id="max_speed"
          name="max_speed"
          placeholder="355 km/h"
          required={true}
          type="text"
        />
      </div>
      <div
        className="add-margin-below"
      >
        <label
          htmlFor="acceleration"
        >
          Acceleration
        </label>
        <input
          className="form-field"
          id="acceleration"
          name="acceleration"
          placeholder="9.8s"
          required={true}
          type="text"
        />
      </div>
    </div>
    <h3>
      Description
    </h3>
    <div
      className="add-margin-below"
    >
      <textarea
        className="form-field"
        name="description"
        required={true}
      />
    </div>
  </fragment>,
  <div
    className="menuMobile"
    onClick={[Function]}
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
  </div>,
  <div
    className="navbar nav-show"
  >
    <div
      className="closeBtn"
      onClick={[Function]}
    >
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </div>
    <div
      className="navTop"
    >
      <a
        className="img-container"
        href="/"
        onClick={[Function]}
      />
      <nav>
        <ul>
          <li>
            <a
              className=""
              href="/models"
              onClick={[Function]}
            >
              MODELS
            </a>
          </li>
          <li>
            <a
              className=""
              href="/reservation"
              onClick={[Function]}
            >
              MY RESERVATIONS
            </a>
          </li>
          <li>
            <a
              className=""
              href="/booking"
              onClick={[Function]}
            >
              RESERVATIONS
            </a>
          </li>
          <li>
            <a
              className=""
              href="/about"
              onClick={[Function]}
            >
              ABOUT
            </a>
          </li>
          <li>
            <a
              className=""
              href="/addCar"
              onClick={[Function]}
            >
              ADD CAR
            </a>
          </li>
          <li>
            <a
              className=""
              href="/deleteCar"
              onClick={[Function]}
            >
              DELETE CAR
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="social-icons"
    >
      <ul>
        <li>
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </li>
        <li>
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            />
          </svg>
        </li>
        <li>
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 640 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M386.061 228.496c1.834 9.692 3.143 19.384 3.143 31.956C389.204 370.205 315.599 448 204.8 448c-106.084 0-192-85.915-192-192s85.916-192 192-192c51.864 0 95.083 18.859 128.611 50.292l-52.126 50.03c-14.145-13.621-39.028-29.599-76.485-29.599-65.484 0-118.92 54.221-118.92 121.277 0 67.056 53.436 121.277 118.92 121.277 75.961 0 104.513-54.745 108.965-82.773H204.8v-66.009h181.261zm185.406 6.437V179.2h-56.001v55.733h-55.733v56.001h55.733v55.733h56.001v-55.733H627.2v-56.001h-55.733z"
            />
          </svg>
        </li>
        <li>
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M204 6.5C101.4 6.5 0 74.9 0 185.6 0 256 39.6 296 63.6 296c9.9 0 15.6-27.6 15.6-35.4 0-9.3-23.7-29.1-23.7-67.8 0-80.4 61.2-137.4 140.4-137.4 68.1 0 118.5 38.7 118.5 109.8 0 53.1-21.3 152.7-90.3 152.7-24.9 0-46.2-18-46.2-43.8 0-37.8 26.4-74.4 26.4-113.4 0-66.2-93.9-54.2-93.9 25.8 0 16.8 2.1 35.4 9.6 50.7-13.8 59.4-42 147.9-42 209.1 0 18.9 2.7 37.5 4.5 56.4 3.4 3.8 1.7 3.4 6.9 1.5 50.4-69 48.6-82.5 71.4-172.8 12.3 23.4 44.1 36 69.3 36 106.2 0 153.9-103.5 153.9-196.8C384 71.3 298.2 6.5 204 6.5z"
            />
          </svg>
        </li>
      </ul>
      <p>
        Microverse Copyright 2022
      </p>
    </div>
  </div>,
]
`;
